### PR TITLE
Docs: export customizations

### PR DESCRIPTION
**Pull request by svdimitr**

### CODE REVIEW

File: components/grid/export/csv.md [hunk 1]
```diff
@@ -14,6 +14,11 @@ You can export the grid to CSV with a click of a button. The current filter, sor
 
 When you click the Export button, your browser will receive the resulting file.
 
+#### In this article
+
+* [Basics](#basics)
+* [Programmatic Export From Code](#programmatic-export-from-code)
```
Comment (dimodi) [inline components/grid/export/csv.md:20]: We say the same thing in two different ways?

```suggestion
* [Programmatic Export](#programmatic-export)
```

File: components/grid/export/csv.md [hunk 2]
```diff
@@ -87,6 +92,90 @@ Optionally, you can also set the `GridCsvExport` tag settings under the `GridExp
 }
 ````
 
+## Programmatic Export From Code
```
Comment (dimodi) [inline components/grid/export/csv.md:95]:
```suggestion
## Programmatic Export
```

File: components/grid/export/csv.md [hunk 3]
```diff
@@ -87,6 +92,90 @@ Optionally, you can also set the `GridCsvExport` tag settings under the `GridExp
...
+
+You can programmatically invoke the export feature of the Grid, by using the following methods exposed on the `@ref` of the Grid:
+
+* `SaveAsExcelFileAsync` - `ValueTask` - sends the exported excel file to the browser for download
```
Comment (dimodi) [inline components/grid/export/csv.md:99]: Not sure if it's clear that the second word is the return type.

File: components/grid/export/events.md [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+    * [Examples](#examples)
+
+* [OnAfterExport](#onafterexport)
+    * [For Excel Export](#for-excel-export)
+    * [For CSV export](#for-csv-export)
+    * [Examples](#examples)
+
+## OnBeforeExport
+
+The `OnBeforeExport` event fires after the user clicked the `ExcelExport` or `CsvExport` buttons, or by [programmatically invoking the export]({%slug grid-export-excel%}#programmatic-export-from-code). The event handler receives a `GridBeforeExcelExportEventArgs` and `GridBeforeCsvExportEventArgs` object, depending on the type of export, which provides the following fields:
```
Comment (dimodi) [inline components/grid/export/events.md:29]:
```suggestion
The `OnBeforeExport` event fires when the user clicks on a export button, or when [export is invoked programmatically]({%slug grid-export-excel%}#programmatic-export). The event handler receives a `GridBeforeExcelExportEventArgs` and `GridBeforeCsvExportEventArgs` object, depending on the type of export, which provides the following fields:
```

File: components/grid/export/events.md [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+
+## For Excel Export
+
+* `Columns` - `List<ExcelExportableColumns` - a collection of all exportable columns in the Grid (the columns that have a defined `Field` and are visible). You can customize the following attributes of the Grid column before exporting it into Excel:
```
Comment (dimodi) [inline components/grid/export/events.md:33]:
```suggestion
* `Columns` - `List<ExcelExportableColumns` - a collection of all exportable columns in the Grid (the columns that have a defined `Field` and are visible). You can customize the following attributes of the Grid columns before exporting to Excel:
```

File: components/grid/export/events.md [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+
+    * `Width` - define the width of the column when exported to excel file.
```
Comment (dimodi) [inline components/grid/export/events.md:35]:
```suggestion
    * `Width` - define the width of the column **in pixels**.
```

File: components/grid/export/events.md [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+    * `Title` - define the title of the column that would be shown in the header in excel file. 
+    * `NumberFormat` - you can use the `NumberFormat` to provide an Excel-compatible number/date format
+    * `Field` - provides the data bound field of the column.
```
Comment (dimodi) [inline components/grid/export/events.md:38]:
```suggestion
    * `Field` - set the data bound field of the column.
```

File: components/grid/export/events.md [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+    
+    
+* `Data` - `IEnumerable<object>` - you can assign a custom collection of data to be exported to excel such as the selected items in the Grid. 
+
+* `isCancelled` -  `bool` - you can cancel the OnBeforeExcel event by setting the `isCancelled` field to `true`.
```
Comment (dimodi) [inline components/grid/export/events.md:43]: field values normally don't change

```suggestion
* `isCancelled` -  `bool` - cancel the OnBeforeExcel event by setting the `isCancelled` property to `true`.
```

File: components/grid/export/events.md [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+
+
+### For CSV Export
+
+* `Data` - `IEnumerable<object>` - you can assign a custom collection of data to be exported to excel/csv such as the selected items in the Grid.
```
Comment (dimodi) [inline components/grid/export/events.md:48]:
```suggestion
* `Data` - `IEnumerable<object>` - assign a custom collection of data to be exported to CSV, for example only the selected items in the Grid. 
```

File: components/grid/export/events.md [hunk 10]
```diff
@@ -0,0 +1,364 @@
...
+* `Data` - `IEnumerable<object>` - you can assign a custom collection of data to be exported to excel such as the selected items in the Grid. 
+
+* `isCancelled` -  `bool` - you can cancel the OnBeforeExcel event by setting the `isCancelled` field to `true`.
+
+
+### For CSV Export
+
+* `Data` - `IEnumerable<object>` - you can assign a custom collection of data to be exported to excel/csv such as the selected items in the Grid. 
+
+* `isCancelled` -  `bool` - you can cancel the OnBeforeExcel event by setting the `isCancelled` field to `true`.
```
Comment (dimodi) [inline components/grid/export/events.md:50]:
```suggestion
* `isCancelled` -  `bool` - cancel the OnBeforeExcel event by setting the `isCancelled` property to `true`.
```

File: components/grid/export/events.md [hunk 11]
```diff
@@ -0,0 +1,364 @@
...
+
+>note Unlike Excel, the CSV files do not have defined columns, so the `args.Columns` field will not be applicable when exporting the Grid to CSV.
+
+
+### Examples
+
+<div class="skip-repl"></div>
+````Excel
+@* This example showcases the capabilities of the OnBeforeExport event when exporting the Grid to excel. *@
```
Comment (dimodi) [inline components/grid/export/events.md:59]:
```suggestion
@* This example shows the capabilities of the OnBeforeExport event when exporting the Grid to excel. *@
```

File: components/grid/export/events.md [hunk 12]
```diff
@@ -0,0 +1,364 @@
...
+        public decimal Price { get; set; }
+        public bool Discontinued { get; set; }
+        public DateTime FirstReleaseDate { get; set; }
+    }
+}
+````
+
+## OnAfterExport
+
+The `OnAfterExport` event fires after the [OnBeforeExport](#onbeforeexport) event, and before the actual file is provided to the user. The event handler receives a `GridAfterExcelExportEventArgs` and `GridAfterCsvExportEventArgs` object, depending on the type of export, which provides the following fields:
```
Comment (dimodi) [inline components/grid/export/events.md:205]:
```suggestion
The `OnAfterExport` event fires after the [OnBeforeExport](#onbeforeexport) event, and before the actual file is provided to the user. The event handler receives a `GridAfterExcelExportEventArgs` or `GridAfterCsvExportEventArgs` object, depending on the type of export, which provides the following fields:
```

File: components/grid/export/excel.md [hunk 13]
```diff
@@ -87,6 +93,91 @@ Optionally, you can also set the `GridExcelExport` tag settings under the `GridE
 }
 ````
 
+## Programmatic Export From Code
```
Comment (dimodi) [inline components/grid/export/excel.md:96]:
```suggestion
## Programmatic Export
```

**Review by dimodi — APPROVED** [review]
Use consistent grammar and structure in list items. For example, no need to use "you can" in some items, and skip it elsewhere. It's best to skip it everywhere, because it does not add value and make the item longer.

**Comment by marin-bratanov** [comment]
Add mentions of the export-related events into a dedicated section in the common Grid Events article, with just a link to the dedicated article, like we have for the CRUD events, OnRead, Column events.

**Pull request merged**